src/repart/repart.c

```diff
@@ -10353,41 +10357,42 @@ static int vl_method_run(
 
         r = context_ponder(context);
         if (r == -ENOSPC) {
+                uint64_t size = UINT64_MAX, current_size = UINT64_MAX;
+
+                r = determine_auto_size(context, LOG_DEBUG, &size, &current_size);
+
                 /* Check if space issue is caused by the whole disk being too small */
-                uint64_t size = UINT64_MAX;
-                (void) determine_auto_size(context, LOG_DEBUG, /* ignore_allocated= */ false, &size);
-                if (size != UINT64_MAX && context->total != UINT64_MAX && size > context->total)
+                if (r == 0 && size > context->total)
                         return sd_varlink_errorbo(
                                         link,
                                         "io.systemd.Repart.DiskTooSmall",
                                         SD_JSON_BUILD_PAIR_UNSIGNED("minimalSizeBytes", size),
-                                        SD_JSON_BUILD_PAIR_UNSIGNED("currentSizeBytes", context->total));
+                                        SD_JSON_BUILD_PAIR_UNSIGNED("currentSizeBytes", current_size));
 
                 /* Or if the disk would fit, but theres's not enough unallocated space */
-                uint64_t need_free = UINT64_MAX;
-                (void) determine_auto_size(context, LOG_DEBUG, /* ignore_allocated= */ true, &need_free);
+                uint64_t need_free = LESS_BY(size, current_size);
```

Comment:
Technically, this is correct, since `size` and `current_size` are initialized. But if `determine_auto_size` fails, going through the code here feels very strange.

Why not do something like this:
```C
if (r == -ENOSPC) {
  uint64_t current_size, new_size;

  r = determine_auto_size(context, LOG_DEBUG, &current_size, &new_size);
  if (r < 0)
     return r;

  if (size > context->total)
     return ...

  uint64_t need_free = LESS_BY(size, current_size);
  return ...
}
```